services/comm.h

```diff
@@ -61,6 +61,54 @@
 #define IS_PROTOCOL_34(c) ((c)->protocol >= 34)
 #define IS_PROTOCOL_35(c) ((c)->protocol >= 35)
 
+
+/*
+ * juliengregoire@outlook.com Patch:
+ * Create a class for reading the environement variables and replace the 
+ * the hardcoded ports !
+ *
+ */
+
+class PortCustomConfig
+{	
+	public:
+	
+	unsigned int daemonport;
+	unsigned int schedulerport;
+		
+	char* dp;
+	char* sp;
+	
+	PortCustomConfig()
+	{
+		if(getenv("daemonport") && getenv("schedulerport") != 0)
+		{
```

Comment:
Why do I have to specify both?  I understand for your use case you specify both, but your patch should be reusable for those who only need to change one (ie something else on their network is using scheduler port already, but the saemon port is free)